tests/auxiliary/mod.rs

```diff
@@ -90,7 +91,13 @@ pub fn normalize_output(output_path: &Utf8Path, args: &[&str]) -> Result<()> {
         if !args.contains(&"--summary-only") {
             json.demangle();
         }
-        fs::write(output_path, serde_json::to_vec_pretty(&json)?)?;
+        let json = if args.contains(&"--codecov") {
+            let codecov = CodeCovJsonExport::from(json);
+            serde_json::to_vec_pretty(&codecov)?
+        } else {
+            serde_json::to_vec_pretty(&json)?
+        };
+        fs::write(output_path, json)?;
```

Comment:
This function is for sanitizing platform-specific paths and demangling function names in the reports generated during testing.

For argument handling and report generation, https://github.com/taiki-e/cargo-llvm-cov/pull/224, which supported Cobertura format in a similar way, may be helpful.
(Basically,  `cli.rs` [handles the arguments](https://github.com/taiki-e/cargo-llvm-cov/pull/224/files#diff-b2812f19576dd53d0c35b107a322f58a00fce6977f4b5976e1961853982af3cc), and `main.rs`'s `Format` enum [handle the format-specific code](https://github.com/taiki-e/cargo-llvm-cov/pull/224/files#diff-42cb6807ad74b3e201c5a7ca98b911c5fa08380e942be6e4ac5807f8377f87fc).)

Reply:
removed this :)